torch/onnx/_internal/fx/_pass.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import abc
+import difflib
+
+from typing import Any, Callable, Dict, Generator, Sequence, Tuple
+
+import torch
+import torch.fx
+from torch.onnx._internal import _beartype
+from torch.onnx._internal.fx import diagnostics
+
+
+class FxReadableGraphDiffer(difflib.Differ):
+    """Differ specialized for fx readable graph.
+
+    The only difference between this and official `difflib.Differ` is the `autojunk`
+    argument for `difflib.SequenceMatcher` class. In this class, `autojunk` is set
+    to `False`. This is to prevent `difflib.SequenceMatcher` recognizing stacktrace
+    messages in fx readable graph as junk, as these messages tend to be long (>200)
+    and repeat multiple times, which falls under the junk filter criteria.
+
+    `Reference: Automatic junk heuristic <https://docs.python.org/3/library/difflib.html>`_
+    """
+
+    def compare(self, a: Sequence[str], b: Sequence[str]) -> Generator[str, None, None]:
+        cruncher = difflib.SequenceMatcher(None, a, b, autojunk=False)
+        for tag, alo, ahi, blo, bhi in cruncher.get_opcodes():
+            if tag == "replace":
+                g = self._fancy_replace(a, alo, ahi, b, blo, bhi)  # type: ignore[attr-defined]
+            elif tag == "delete":
+                g = self._dump("-", a, alo, ahi)  # type: ignore[attr-defined]
+            elif tag == "insert":
+                g = self._dump("+", b, blo, bhi)  # type: ignore[attr-defined]
+            elif tag == "equal":
+                g = self._dump(" ", a, alo, ahi)  # type: ignore[attr-defined]
+            else:
+                raise ValueError("unknown tag %r" % (tag,))
+
+            yield from g
+
+
+class Pass(abc.ABC):
```

Comment:
What is this for? Can we remove it and  do
```
class Transform(abc.ABC))
```

Reply:
For future analysis, but you are right for the context of this PR it makes sense to just use Transform as base.